### PR TITLE
[Qt] prepare paymentservertests for new unit tests

### PR DESCRIPTION
**Pull request by Diapolo**
The intention of this pull is to be able to extend the existing unit-tests for payment requests. This is achieved by adding a second `PaymentRequest Test CA` (serial number f0:da:97:e4:38:d7:64:16) into paymentrequestdata.h. The CA was generated by my using "ca_in_a_box" (by @gavinandresen), which I've partly re-worked to natively run on Windows. Also the code in paymentservertests was reworked to be able to directly load payment request raw data from paymentrequestdata.h into an usable `PaymentRequestPlus` object. This is used to skip normal client code flow, to be able to explicitly test and verify payment request validity.

As a first step I integrated a new unit test, which is able to verify that the "client network matches the payment request network" rule is working as intended.

### CODE REVIEW

**Comment by Diapolo** [comment]
@gavinandresen @laanwj Travis should pass now and I'm now able to extend unit-tests after this is merged. Can you review and/or comment this? I know there are quite a few open payment request related pulls, but hey you wanted me to to something more valuable than just cleanups ^^.

**Comment by gavinandresen** [comment]
Untested ACK. You should edit the first comment in this pull request to describe what the pull request does.

**Comment by Diapolo** [comment]
@gavinandresen Thanks, I updated the initial comment. Would be great if we could get that in soon :).

File: src/qt/paymentserver.cpp [hunk 1]
```diff
@@ -745,3 +744,16 @@ void PaymentServer::handlePaymentACK(const QString& paymentACKMsg)
     // currently we don't futher process or store the paymentACK message
     emit message(tr("Payment acknowledged"), paymentACKMsg, CClientUIInterface::ICON_INFORMATION | CClientUIInterface::MODAL);
 }
+
+bool PaymentServer::verifyNetwork(const payments::PaymentDetails& requestDetails)
+{
+    const QString requestNetwork = QString::fromStdString(requestDetails.network());
```
Comment (laanwj) [inline src/qt/paymentserver.cpp]: Why change these to QString just to compare?

Comment (Diapolo) [inline src/qt/paymentserver.cpp]: @laanwj The logging only works when converted to QString ;).

Comment (Diapolo) [inline src/qt/paymentserver.cpp]: @laanwj If you want I can change this into:
<pre>
    bool fVerified = requestDetails.network() == Params().NetworkIDString();
    if (!fVerified) {
        qWarning() << QString("PaymentServer::%1: Payment request network \"%2\" doesn't match client network \"%3\".")
            .arg(__func__)
            .arg(QString::fromStdString(requestDetails.network()))
            .arg(QString::fromStdString(Params().NetworkIDString()));
    }
    return fVerified;
</pre>

Comment (laanwj) [inline src/qt/paymentserver.cpp]: Yes. That's better, just convert as needed.

**Comment by laanwj** [comment]
ACK, more tests is good.

**Comment by Diapolo** [comment]
@laanwj Great, that counts 2 ACKs now. The last rebase just changed ``PaymentServer::verifyNetwork()`` to only convert networks as needed.

I'll be happy to add aditional testing for payment requests :).

**Pull request merged**